### PR TITLE
Flexible Classroom 2.8.0 RN & API updates

### PR DESCRIPTION
**Pull request by atovpeko**

### CODE REVIEW

File: shared/flexible-classroom/classroom-sdk/android.mdx [hunk 1]
```diff
@@ -374,12 +374,12 @@ The classroom type. Set in [AgoraEduLaunchConfig](#agoraedulaunchconfig).
...
-| `AgoraEduRoomTypeBig` | `2`: Lecture Hall. A teacher gives an online lesson to multiple students. Students do not send their audio and video by default. There is no upper limit on the number of students. During the class, students can "raise their hands" to apply for speaking up. Once the teacher approves, the student can send their audio and video to interact with the teacher. |
-| `AgoraEduRoomTypeSmall` | `4`: Small Classroom. A teacher gives an online lesson to multiple students. Students do not send their audio and video by default. The maximum number of users in a classroom is 500. During the class, the teacher can invite students to speak up "on stage" and have real-time audio and video interactions with the teacher. |
+| `AgoraEduRoomTypeBig` | `2`: Lecture Hall. A teacher gives an online lesson to multiple students. Students do not send their audio and video by default. The maximum number of users in a classroom is 5,000. During the class, students can "raise their hands" to apply for speaking up. Once the teacher approves, the student can send their audio and video to interact with the teacher. |
```
Comment (billy-the-fish) [inline shared/flexible-classroom/classroom-sdk/android.mdx:377]: Can you get rid of the "quotation fingers" please. 

Please change `to apply for speaking up` into something a bit clearer.

File: shared/flexible-classroom/classroom-sdk/android.mdx [hunk 2]
```diff
@@ -374,12 +374,12 @@ The classroom type. Set in [AgoraEduLaunchConfig](#agoraedulaunchconfig).
...
+| `AgoraEduRoomTypeSmall` | `4`: Small Classroom. A teacher gives an online lesson to multiple students. Students do not send their audio and video by default. The maximum number of users in a classroom is 200. During the class, the teacher can invite students to speak up "on stage" and have real-time audio and video interactions with the teacher. |
```
Comment (billy-the-fish) [inline shared/flexible-classroom/classroom-sdk/android.mdx:378]: "quotation fingers"

File: shared/flexible-classroom/classroom-sdk/ios.mdx [hunk 3]
```diff
@@ -261,8 +261,8 @@ The classroom type. Set in [AgoraEduLaunchConfig](#agoraedulaunchconfig).
...
-| `AgoraEduRoomTypeBig` | `2`: Lecture Hall. A teacher gives an online lesson to multiple students. Students do not send their audio and video by default. There is no upper limit on the number of students. During the class, students can "raise their hands" to apply for speaking up. Once the teacher approves, the student can send their audio and video to interact with the teacher. |
-| `AgoraEduRoomTypeSmall` | `4`: Small Classroom. A teacher gives an online lesson to multiple students. Students do not send their audio and video by default. The maximum number of users in a classroom is 500. During the class, the teacher can invite students to speak up "on stage" and have real-time audio and video interactions with the teacher. |
+| `AgoraEduRoomTypeBig` | `2`: Lecture Hall. A teacher gives an online lesson to multiple students. Students do not send their audio and video by default. The maximum number of users in a classroom is 5,000. During the class, students can "raise their hands" to apply for speaking up. Once the teacher approves, the student can send their audio and video to interact with the teacher. |
```
Comment (billy-the-fish) [inline shared/flexible-classroom/classroom-sdk/ios.mdx:264]: Same as previous comments.  

Is there any difference between the APIs for the different platforms?

File: shared/flexible-classroom/classroom-sdk/js.mdx [hunk 4]
```diff
@@ -153,25 +153,33 @@ export type LaunchOption = {
...
 | `pretest` | (Required) Whether to enable the pre-class device test:<li>`true`: Enable the pre-class device test. After this function is enabled, end users can see a page for the device test before entering the classroom. They can check whether their camera, microphone, and speaker can work properly.</li><li>`false`: Disable the pre-class device test.</li> |
 | `language` | (Required) The UI language. See [`LanguageEnum`](#languageenum). |
 | `startTime` | (Required) The start time (ms) of the class, determined by the first user joining the classroom. |
 | `duration` | (Required) The duration (second) of the class, determined by the first user joining the classroom. |
 | `recordUrl` | (Optional) The URL address to be recorded. Developers need to pass in the URL of the web page deployed by themselves for page recording, such as `https://cn.bing.com/recordUrl`. |
+| `widgets`| (Optional) Extensive widgets that  extend the classroom capabilities. See [Embed a custom plugin](https://docs.agora.io/en/flexible-classroom/develop/embed-custom-plugin?platform=webEmbed) for details.|
```
Comment (billy-the-fish) [inline shared/flexible-classroom/classroom-sdk/js.mdx:163]: we don't have a platform called webEmbed

File: shared/flexible-classroom/classroom-sdk/js.mdx [hunk 5]
```diff
@@ -153,25 +153,33 @@ export type LaunchOption = {
...
 | `latencyLevel` | (Optional) The latency level of an audience member in interactive live streaming:<li>`1`: Low latency. The latency from the sender to the receiver is 1500 ms to 2000 ms.</li><li>(Default) Ultra-low latency. The latency from the sender to the receiver is 400 ms to 800 ms.</li> |
+| `virtualBackgroundImages` | (Optional) The URL of the virtual background image. The URL's domain must be the same as `webrtcExtensionBaseUrl`. The PNG and JPG formats are supported. |
+| `webrtcExtensionBaseUrl` | (Optional) The URL or the WebRtc extensions. The default value is `https://solutions-apaas.agora.io/static`. If you want to use the advanced features such as virtual backgrounds, AI noise suppression, and beauty options, you need ti implement the WebRtc extensions and relevant resources in the domain of Flexible Classroom SDK. These are the steps: When you run `yarn build:demo` to complete packaging, the corresponding files are generated in `packages/agora-demo-app/build/extensions`. The you implement the directory in the domain of Flexible Classroom SDK. |
```
Comment (billy-the-fish) [inline shared/flexible-classroom/classroom-sdk/js.mdx:169]: Can you have a quick glance over this sentence please.

File: shared/flexible-classroom/proctor-sdk/ios.mdx [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+| `roomUuid` | The room ID. This is the globally unique identifier of a classroom. This value must be less than 64 bytes. The following characters are supported:<ul><li>All lowercase English letters: a to z.</li><li>All uppercase English letters: A to Z.</li><li>All numeric characters: 0-9.</li><li>The space character.</li><li> `"!"`, `"#"`, `"$"`, `"%"`, `"&"`, `"("`, `")"`, `"+"`, `"-"`, `":"`, `";"`, `"<"`, `"="`, `"."`, `">"`, `"?"`, `"@"`, `"["`, `"]"`, `"^"`, `"_"`, `"{"`, `"}"`, `"\|"`, `"~"`, `","`</li></ul> |
+| `appId`          | The Agora App ID. See [Get the Agora App ID](https://docs.agora.io/en/flexible-classroom/reference/manage-agora-account?platform=ios#get-the-app-id). |
+| `token` | The signaling token used for authentication. For details, see [Generate a signaling Token](https://docs.agora.io/en/signaling/develop/authentication-workflow). |
+| `region` | The region where the classrooms is. All clients must use the same region; otherwise, they may fail to communicate with each other. The following regions are supported:<ul><li>`CN`: Mainland China</li><li>`NA`: North America</li><li>`EU`: Europe</li><li>`AP`: Asia Pacific</li></ul> |
```
Comment (billy-the-fish) [inline shared/flexible-classroom/proctor-sdk/ios.mdx:135]: classrooms are

 All clients must use the same region. The following ...

File: shared/flexible-classroom/proctor-sdk/ios.mdx [hunk 7]
```diff
@@ -0,0 +1,279 @@
...
+                               bitRate:(NSUInteger)bitRate
+                            mirrorMode:(AgoraProctorMirrorMode)mirrorMode;
+```
+
+The video encoder configuration used in [AgoraProctorMediaOptions](#agoraproctormediaoptions).
+
+
+| Property              | Description                                                           |
+| :---------------- | :------------------------------------------------------------- |
+| `dimensionWidth`  | The width (pixels) of the video frame. The default value is 320.  |
```
Comment (billy-the-fish) [inline shared/flexible-classroom/proctor-sdk/ios.mdx:204]: width in pixels of ... .

Same for all.

File: shared/flexible-classroom/proctor-sdk/js.mdx [hunk 8]
```diff
@@ -0,0 +1,281 @@
...
+```typescript
+export type ConfigParams = {
+    appId: string;
+    region?: string;
+};
+```
+
+| Property     | Description                                   |
+| :------- | :-------------------------------------------------------- |
+| `appId`  | The Agora App ID. See [Get the Agora App ID](https://docs.agora.io/en/flexible-classroom/reference/manage-agora-account?platform=ios#get-the-app-id).   |
```
Comment (billy-the-fish) [inline shared/flexible-classroom/proctor-sdk/js.mdx:52]: Please don't use the full URL. Same for any links in here.

File: shared/flexible-classroom/release-notes/android.mdx [hunk 9]
```diff
@@ -1,5 +1,32 @@
 <PlatformWrapper platform="android">
 
+## v2.8.0
```
Comment (billy-the-fish) [inline shared/flexible-classroom/release-notes/android.mdx:3]: Looks like all the release notes are the same. Please just use one file.

**Review by billy-the-fish — CHANGES_REQUESTED** [review]
Nearly there.

File: shared/flexible-classroom/release-notes/android.mdx (hunk 9 above)
Comment (atovpeko) [reply shared/flexible-classroom/release-notes/android.mdx:3]: They are not

File: shared/flexible-classroom/classroom-sdk/ios.mdx (hunk 3 above)
Comment (atovpeko) [reply shared/flexible-classroom/classroom-sdk/ios.mdx:264]: Yes, there are.

File: shared/flexible-classroom/release-notes/android.mdx (hunk 9 above)
Comment (billy-the-fish) [reply shared/flexible-classroom/release-notes/android.mdx:3]: :-)

**Review by billy-the-fish — APPROVED** [review]

**Pull request merged**